src/week2/index.md

```diff
@@ -417,6 +410,62 @@ Finally you can run the file by executing to see the evaluated output:
 $ ./test/add1.run
 131
 ```
+- Note: Locally on arm macs, to target with Rosetta 2, you need to use:
+```
+cargo run --target x86_64-apple-darwin -- -c test.snek test.s
+```
+and
+```
+nasm -f macho64
+ar rcs runtime/libour_code.a runtime/our_code.o
+rustc --target x86_64-apple-darwin -L tests/ -lour_code:$* runtime/start.rs -o tests/$*.run
+```
+
+## Part 2: Dynamic Compilation
+Now that you have an AOT compiler with all of the Boa language features, you can use it to generate machine code and execute it at runtime. Just like in Adder, you will use the [dynasm](https://github.com/CensoredUsername/dynasm-rs) crate to generate machine code at runtime. 
+
+The dynasm crate provides macros that allow you to write assembly-like code that gets compiled into a vector of bytes representing machine instructions. You can then execute this machine code by casting it to a function pointer using `mem::transmute`. This is exactly how you evaluate the Boa program directly without generating an assembly file or linking it to `runtime.rs`!
+
+### Writing the JIT Compiler
+Notice how we used the `Instr` type to represent assembly instructions in Part 1. This abstraction is useful for generating assembly code as a string. But how about for JIT compilation, where we need to generate machine code directly with dynasm's macros? Using a vector of `Instr`, instead of generating strings, is it possible to generate machine code directly? It may be useful to have a function like this:
+```rust
+fn instr_to_asm(i: &Instr, ops: &mut dynasmrt::x64::Assembler) {
+    todo!("instr_to_asm");
+}
+```
+It would be a good idea to have a thorough understanding of all possible `Instr` variants and their corresponding assembly instructions. Although this is one approach, feel free to use any strategy you would like!
+
+### Running
+
+Now with the `-e` flag, run `cargo run -- -e test.snek` to compile your snek file and directly execute it. You can modify your `Makefile` to also include this command.
+
+```
+$ make -e test/add1.snek
+cargo run -e -- test/add1.snek
+131
+```
+
+Here we have already evaluated our program at runtime! Notice that we did not link it to `runtime.rs` to create an executable. 
+
+We also will include a `-g` with `cargo run -- -g test.snek test.s` flag that will do both AOT and JIT compilation. This is for debugging purposes, since we can evaluate Boa code directly at runtime and print out the (hopefully correct) generated assembly that should produce the same evaluation. In all, your AOT and JIT compilers should be producing the same results.
+
+```
+$ make -g test/add1.s
```

Comment:
Huh I didn't know you could give the `-g` flag through to the underlying command like this. How does that work?